product-service/src/libs/data-service.ts

```diff
@@ -1,10 +1,118 @@
-import { productsData } from '@mocks/products';
-import { IProduct, IProductsData } from 'src/types/api-types';
+import { Product, ProductsData, Stock } from 'src/types/api-types';
+import {
+  AttributeValue,
+  DynamoDBClient,
+  PutItemCommand,
+  PutItemCommandInput,
+  QueryCommand,
+  QueryInput,
+  ScanCommand,
+  ScanCommandInput
+} from '@aws-sdk/client-dynamodb';
 
-export const getProducts = async (): Promise<IProductsData> => {
-  return Promise.resolve(productsData);
+const dynamodb = new DynamoDBClient({ region: process.env.REGION || 'us-east-1' });
+
+function deserializeDynamoDBItem(item: Record<string, AttributeValue>) {
+  const deserializedItem = {};
+  for (const [attributeName, value] of Object.entries(item)) {
+    const attributeType = Object.keys(value)[0];
+
+    if (attributeType === 'S') {
+      deserializedItem[attributeName] = value.S;
+    } else if (attributeType === 'N') {
```

Comment:
**Concern**:  What happens if we get attributeType === 'M'  or other? - we don't get value in response.
Consider to find another approach.

Reply:
I guess it should solve your issue:
`const client = DynamoDBDocumentClient.from(new DynamoDBClient({ region: process.env.REGION ?? 'eu-central-1' }));`

Reply:
![Uploading image.png…]()

Reply:
wow thanks I'll try this out